pkg/updater/repository/link_test.go

```diff
@@ -0,0 +1,92 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build !windows
+
+package repository
+
+import (
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
```

Comment:
You might want to use `require` instead of `assert`, with `assert` the test isn't stopped if something fails so you might get unexpected errors later on in the test

Reply:
If you don't have a strong opinion I'll keep the asserts, I feel like this is mostly personal preference and at least at Datadog we don't seem to strongly prefer one of the two. IMO reasoning about whether `require` vs `assert` should be used in any particular situation is a bit tricky so I just default to `assert` to have more context from CI when that's useful

Reply:
I use both depending on the situation: for example in the test linked below `os.Open` might fail but it won't stop `assert.NoError`, it will keep going, run `defer f.Close()` and `io.Copy(hash, f)`, which are both likely invalid

https://github.com/DataDog/datadog-agent/blob/957202f40fab962f66956466d76ada6df1632fcc/pkg/updater/download_test.go#L44-L52

The proper way to handle that if you want to use `assert` is to check the boolean returned by `NoError` to know if there was actually and error.

All registered errors will still appear when running the test so you just need to know that you should only look at the first one, but I find it less clear than just using `require` in that case.